src/ui/public/kuery/lib/functions/not.js

```diff
@@ -0,0 +1,38 @@
+import * as ast from '../ast';
+import { nodeTypes } from '../node_types';
+
+export function buildNodeParams(child, serializeStyle = 'operator') {
+  return {
+    arguments: [child],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  let [ argument ] = node.arguments;
+  if (argument.type === 'literal') {
+    argument = nodeTypes.function.buildNode('is', '*', argument.value);
+  }
+
+  return {
+    bool: {
+      must_not: ast.toElasticsearchQuery(argument, indexPattern)
+    }
+  };
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
```

Comment:
`:[ to ]` just seemed weird to me... it felt wrong to call it an operator since it's not strictly prefix or infix. However, after giving it some more thought and perusing the [Operator Wikipedia page](https://en.wikipedia.org/wiki/Operator_(computer_programming)), maybe it's ok. It's not much weirder than the ternary operator. I thought calling it `shorthand` instead of `operator` would reduce confusion, but it seems that's not the case ;)